docs/source/notes/autograd.rst

```diff
@@ -944,3 +956,9 @@ of that Tensor, so if that Tensor is then modified in-place,
 even though the Tensor now has a new grad_fn, hooks registered before it was
 modified in-place will continue to be associated with the old grad_fn, e.g. they will
 fire when that Tensor's old grad_fn is reached in the graph by the autograd engine.
+
+The one exception is post accumulate grad hooks, which receive the tensor that requires grad
+instead of the gradient and expect users to modify the tensor in-place. Ideally, the post
+grad accumulate hooks should be last to execute in the backward pass. The only contention
+would occur if you also registered a post-hook on the AccumulateGrad node, which is not
+recommended, and in that case, it is up to you to ensure sanity.
```

Comment:
I should definitely rewrite this to be like "be cautious and set torch.no_grad for content in the hook if you don't want unforeseen circumstances" and get rid of the "it's last so should be okay". Being last does not make it okay since people run many fwd/bwd passes

Reply:
Note that differentiable inplace on leaf is not allowed. So this comment is moot as you will never be able to register such hook!

Reply:
So technically...it is "allowed" if  the person enables torch.grad right? Even then, this chunk shouldn't apply to my hooks because I literally give them the Tensor at the time, so any updates from hooks will be reflected.